geomstats/geometry/complex_poincare_disk.py

```diff
@@ -0,0 +1,360 @@
+
+        Returns
+        -------
+        exp : array-like, shape=[..., 1]
+            Riemannian exponential.
+        """
+        ndim = max(len(gs.shape(tangent_vec)), len(gs.shape(base_point)))
+        tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=2)
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        if base_point.shape[0] < tangent_vec.shape[0]:
+            base_point = (
+                gs.cast(gs.ones(tangent_vec.shape), dtype=base_point.dtype) @ base_point
+            )
+        elif base_point.shape[0] > tangent_vec.shape[0]:
+            tangent_vec = (
+                gs.cast(gs.ones(base_point.shape), dtype=tangent_vec.dtype)
+                @ tangent_vec
+            )
+        theta = gs.cast(gs.angle(tangent_vec), dtype=gs.get_default_cdtype())
+        s = 2 * gs.abs(tangent_vec) / (1 - gs.abs(base_point) ** 2)
+        num = gs.array(base_point)
+        exp_i_theta = gs.exp(1j * theta)
+        exp_minus_s = gs.cast(gs.exp(-s), dtype=gs.get_default_cdtype())
+        num += exp_i_theta
+        num += (base_point - exp_i_theta) * exp_minus_s
+        den = 1
+        den += gs.conj(base_point) * exp_i_theta
+        den += (1 - gs.conj(base_point) * exp_i_theta) * exp_minus_s
+        exp = num / den
+        if ndim == 1:
+            exp = gs.reshape(exp, (-1,))
+        return exp
+
+    @staticmethod
+    def _tau(point_a, point_b, atol=gs.atol):
+        """Compute the coefficient tau.
+
+        The coefficient tau is used to compute the distance
+        between point_a and point_b; it is also used to
+        compute the logarithm map between point_a and point_b.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., 1]
+            Point.
+        point_b : array-like, shape=[..., 1]
+            Point.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        tau : array-like, shape=[...,]
+            Coefficient tau.
+        """
+        num = gs.abs(point_a - point_b)
+        den = gs.abs(1 - point_a * gs.conj(point_b))
+        den = gs.maximum(den, atol)
+        delta = num / den
+        delta = gs.minimum(delta, 1 - atol)
+        tau = (1 / 2) * gs.log((1 + delta) / (1 - delta))
+        return tau
+
+    def log(self, point, base_point):
+        """Compute the complex Poincaré disk logarithm map.
+
+        Compute the Riemannian logarithm at point base_point,
+        of point wrt the positive reals metric.
+        This gives a tangent vector at point base_point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point.
+        base_point : array-like, shape=[..., 1]
+            Base point.
+
+        Returns
+        -------
+        log : array-like, shape=[..., 1]
+            Riemannian logarithm.
+        """
+        log = self._tau(base_point, point)
+        log = gs.cast(log, dtype=gs.get_default_cdtype())
```

Comment:
unnecessary `cast`.

Reply:
If I do not cast, TensorFlow gives me the following error:

`tensorflow.python.framework.errors_impl.InvalidArgumentError: cannot compute Mul as input #1(zero-based) was expected to be a double tensor but is a complex128 tensor [Op:Mul]`